src/share/classes/com/sun/javatest/regtest/tool/Tool.java

```diff
@@ -1203,6 +1219,17 @@ public int run() throws BadArgs, Fault, Harness.Fault, InterruptedException {
             reportDirArg = Path.of("JTreport");
         }
 
+        if (customMainWrapper != null) {
+            CustomMainWrapper cmw = CustomMainWrapper.getInstance(customMainWrapper, customMainWrapperPathArg);
+            testVMOpts.add("-D" + MainWrapper.MAIN_WRAPPER + "=" + customMainWrapper);
+            testVMOpts.addAll(cmw.getAdditionalVMOpts());
+        }
+
+        if (customMainWrapperPathArg != null) {
+            testVMOpts.add("-D" + MainWrapper.MAIN_WRAPPER_PATH + "=" + customMainWrapperPathArg);
+        }
```

Comment:
It seems premature to be stashing values in the `testVMOpts` here.  

The more conventional way would be to pass these values through the `RegressionParameters` object to where the instance of the `RegressionScript` can decide whether to take these values into account for each individual test.